### PR TITLE
build auth service backend

### PR DESCRIPTION
**Pull request by Skeletrox**
- modified the exports of dbsdk/index.js

### CODE REVIEW

File: devmgmtV2/controllers/auth.controller.js [hunk 1]
```diff
@@ -1,9 +1,119 @@
...
+ * The getPerms function returns either the list of permission [if a valid user exists],
+ * null otherwise.
+ */
+
+let authLogin = (req,res) => {
+  let userName = req.body['uname'];
+  let password = req.body['pword'];
+  let result = true;
+  verifyIfUserExists(userName).then(response => {
+    if (typeof response[0] === "undefined" || response[0]['password'] != password) {
```
[the remark this answers is not in the document]
Comment (Skeletrox) [reply devmgmtV2/controllers/auth.controller.js]: Around line 99

File: devmgmtV2/controllers/auth.controller.js [hunk 2]
```diff
@@ -1,9 +1,119 @@
...
+    });
+}
+
+let updatePW = (req, res) => {
+  let userName = req.body['uname'];
+  let newPassword = req.body['pword'];
+  let success = null;
+  verifyIfUserExists(userName).then(response => {
+    if (typeof response[0] === "undefined") {
+      res.json({'isSuccessful' : false});
```
[the remark this answers is not in the document]
Comment (Skeletrox) [reply devmgmtV2/controllers/auth.controller.js]: Wouldn't it be more prudent for the frontend to be able to customize whatever they want to show?

File: devmgmtV2/controllers/auth.controller.js [hunk 3]
```diff
@@ -1,9 +1,119 @@
...
+let defaultTableName = 'users';
+let defaultPassword = '';
+
+
+function getColumnFromDB(name, column) {
+  let queryObject = {
+    dbName : defaultDbName,
+    tableName : defaultTableName,
+    fields : [column],
+    where : "where username = '" + name + "'"
```
[the remark this answers is not in the document]
Comment (Skeletrox) [reply devmgmtV2/controllers/auth.controller.js]: MYSQL throws an error on using backticks

File: devmgmtV2/controllers/auth.controller.js [hunk 4]
```diff
@@ -35,85 +35,54 @@ function verifyIfUserExists(name) {
...
-  let result = true;
   verifyIfUserExists(userName).then(response => {
-    if (typeof response[0] === "undefined" || response[0]['password'] != password) {
-      result = false;
-    }
-    res.json({'isSuccessful' : result});
-    }, error => {
-      console.log(error);
-      res.json({'isSuccessful' : null});
+    return new Promise((resolve, reject) => {
```
Comment (rishabh1403) [inline devmgmtV2/controllers/auth.controller.js]: use q library like i have used in db sdk, native promises are not very relaiable now

File: devmgmtV2/controllers/auth.controller.js [hunk 5]
```diff
@@ -35,85 +35,54 @@ function verifyIfUserExists(name) {
...
+      }
+      else {
+        resolve(getColumnFromDB(userName, 'permission'));
+      }
     });
+  }, reject => {
+    console.log(reject);
+    res.status(500).json({authSuccessful : false, authMessage : "Internal server error! Possible SQL Injection attempt"});
+  }).then(response => {
+    let permissionList = JSON.parse(response[0]['permission']);
```
Comment (rishabh1403) [inline devmgmtV2/controllers/auth.controller.js]: front end will do it, you don't handle this, if this break ... your server will break

File: devmgmtV2/controllers/auth.controller.js [hunk 6]
```diff
@@ -35,85 +35,54 @@ function verifyIfUserExists(name) {
...
+      if (response[0] === "undefined" || response[0]['password'] != password) {
+        reject(false);
+      }
+      else {
+        resolve(getColumnFromDB(userName, 'permission'));
+      }
     });
+  }, reject => {
+    console.log(reject);
+    res.status(500).json({authSuccessful : false, authMessage : "Internal server error! Possible SQL Injection attempt"});
```
Comment (rishabh1403) [inline devmgmtV2/controllers/auth.controller.js]: message should not expose our db mechanism

File: devmgmtV2/controllers/auth.controller.js [hunk 7]
```diff
@@ -35,85 +35,54 @@ function verifyIfUserExists(name) {
...
+  }).then(response => {
+    let permissionList = JSON.parse(response[0]['permission']);
+    res.status(200).json({authSuccessful : true, permissions : permissionList, authMessage : "Login Success"});
```
Comment (rishabh1403) [inline devmgmtV2/controllers/auth.controller.js]: send username tooo

File: devmgmtV2/controllers/auth.controller.js [hunk 8]
```diff
@@ -35,85 +35,54 @@ function verifyIfUserExists(name) {
...
+  }, reject => {
+    res.status(401).json({authSuccessful : false, authMessage : "Invalid Username/Password"});
+  });
 }
 
 let updatePW = (req, res) => {
   let userName = req.body['uname'];
   let newPassword = req.body['pword'];
```
Comment (rishabh1403) [inline devmgmtV2/controllers/auth.controller.js]: updatePassword

File: devmgmtV2/controllers/auth.controller.js [hunk 9]
```diff
@@ -35,85 +35,54 @@ function verifyIfUserExists(name) {
...
+  }).then(response => {
+    let permissionList = JSON.parse(response[0]['permission']);
+    res.status(200).json({authSuccessful : true, permissions : permissionList, authMessage : "Login Success"});
+  }, reject => {
+    res.status(401).json({authSuccessful : false, authMessage : "Invalid Username/Password"});
+  });
 }
 
 let updatePW = (req, res) => {
   let userName = req.body['uname'];
```
Comment (rishabh1403) [inline devmgmtV2/controllers/auth.controller.js]: use complete words username, password

File: devmgmtV2/controllers/auth.controller.js [hunk 10]
```diff
@@ -35,85 +35,54 @@ function verifyIfUserExists(name) {
...
-        res.json({permissions:permissionList});
-      }, error => {
-        console.log(error);
-        res.json({permissions : null});
-      });
-    }
-    else {
-      res.json({permissions : null});
-    }
+    return new Promise((resolve, reject) => {
```
Comment (rishabh1403) [inline devmgmtV2/controllers/auth.controller.js]: use q

File: devmgmtV2/controllers/auth.controller.js [hunk 11]
```diff
@@ -35,85 +35,54 @@ function verifyIfUserExists(name) {
...
+      if (typeof response[0] === "undefined") {
+        reject(false);
+      }
+      else {
+        resolve(true);
+      }
+    });
+  }, reject => {
+    res.status(500).json({updateSuccessful : false, updateMessage : "Internal server error! Possible SQL Injection attempt"});
```
Comment (rishabh1403) [inline devmgmtV2/controllers/auth.controller.js]: fix mesaage

File: devmgmtV2/writeToDB.js [hunk 12]
```diff
@@ -0,0 +1,8 @@
+let dbsdk = require('dbsdk');
```
Comment (rishabh1403) [inline devmgmtV2/writeToDB.js]: use destructuring

**Pull request closed without merging**